udata/api/__init__.py

```diff
@@ -120,13 +120,19 @@ def search_parser(self, adapter, paginate=True):
         # q parameter
         parser.add_argument('q', type=str, location='args',
                             help='The search query')
+        facets = adapter.facets.keys()
+        if facets:
+            parser.add_argument('facets', type=str, location='args',
+                                choices=['all'] + facets, action='append',
+                                help='Selected facets to fetch')
         # Add facets filters arguments
         for name, facet in adapter.facets.items():
             parser.add_argument(name, type=str, location='args')
         # Sort arguments
-        choices = (adapter.sorts.keys()
-                   + ['-' + k for k in adapter.sorts.keys()])
-        parser.add_argument('sort', type=str, location='args', choices=choices)
+        keys = adapter.sorts.keys()
+        choices = keys + ['-' + k for k in adapter.sorts.keys()]
```

Comment:
lol, I just understood. You're right, that was the initial point of storing the `keys` variable.